components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -1106,6 +1106,22 @@ impl Into<FontStretch> for f64 {
 impl<H, V> RepeatableListAnimatable for generic_position::Position<H, V>
     where H: RepeatableListAnimatable, V: RepeatableListAnimatable {}
 
+/// https://drafts.csswg.org/css-transitions/#animtype-rect
+impl Animate for ClipRect {
+    #[inline]
+    fn animate(&self, other: &Self, procedure: Procedure) -> Result<Self, ()> {
+        if !self.can_animate_with(other, procedure) {
+            return Err(());
+        }
```

Comment:
There is a better way to do that:

```rust
let animate_component = |this, other| {
    match (this.animate(other, procedure)?, procedure) {
        (None, Procedure::Interpolate { .. }) => Ok(None),
        (None, _) => Err(()),
        (result, _) => Ok(result),
    }
}
```

This way you don't need code to check whether the two values should be allowed to interpolate at all beforehand.

You could even wrap the `Option<T>` of the `ClipRect` fields in its own new type and implement `Animate` by hand for it, keeping the derived code for `ClipRect`.

Reply:
Thanks.
Ah! OK. I'll use Closure!